### PR TITLE
Add Ka-50 ABRIS

### PR DESCRIPTION
**Pull request by okopanja**
Here is the draft pull request.
Major things: 
- extended communication protocol (added self for current coordinates and x, z to coords + added mode reporting for Ka-50)
- extended TheWay.lua to send more data (and use JSON:encode)
- implemented ABRIS entry
- implemented PVI steer point entry
- added definition to pom.xml to enable creation of jar with dependencies, exe and bundle zip ( I did not include jre, this is up to you). 

Please try it.

Testing progress
- [ ] Ka-50:
  - [ ] JJay to test Ka-50. 
    - [ ] PVI steerpoint does not work on machine producing 20 FPS, the PVI enters into unknown mode
  - [x] sprool: everything works
  - [x] okopanja: ok
- [x] Mirage
  - [x] JJay: ok
- [x] F-16
  - [x] okopanja: test shows that F-16 still enters the waypoints.
- [ ] F-18 
  - [ ] ?
- [ ] Apache
  - [ ] ? 
- [x] A-10C
  - [x] sprool: ok 
- [x] A-10C2
  - [x] sprool: ok 
- [x] Harrier
  - [x] sprool: ok

### CODE REVIEW

File: src/main/java/main/Waypoints/PlanesCommands/Ka50.java [hunk 1]
```diff
@@ -155,9 +156,287 @@ public static JSONArray getCommands(List<Point> coords) {
...
+            commandArray.put(new JSONObject().put("device", "20").put("code", "3018").put("delay", "0").put("activate", "10").put("addDepress", "true"));
+        }
+        // PVI: activate steerpoint 1
+        commandArray.put(new JSONObject().put("device", "20").put("code", Integer.toString(3002)).put("delay", "0").put("activate", "1").put("addDepress", "true"));
+        //Press waypoint button
+        commandArray.put(new JSONObject().put("device", "20").put("code", "3011").put("delay", "0").put("activate", "1").put("addDepress", "true"));
+        //Press waypoint button
+        commandArray.put(new JSONObject().put("device", "20").put("code", "3011").put("delay", "0").put("activate", "1").put("addDepress", "true"));
+        // PVI: activate steerpoint 1
+        commandArray.put(new JSONObject().put("device", "20").put("code", Integer.toString(3002)).put("delay", "0").put("activate", "1").put("addDepress", "true"));
```
Comment (okopanja) [inline src/main/java/main/Waypoints/PlanesCommands/Ka50.java:173]: Entering of steer points + activation of flight plan

File: src/main/java/main/Waypoints/PlanesCommands/Ka50.java [hunk 2]
```diff
@@ -155,9 +156,287 @@ public static JSONArray getCommands(List<Point> coords) {
         //PVI to OPER
         commandArray.put(new JSONObject().put("device", "20").put("code", "3026").put("delay", "0").put("activate", "0.3").put("addDepress", "false"));
```
Comment (okopanja) [inline src/main/java/main/Waypoints/PlanesCommands/Ka50.java:158]: Entering of steer points + activation of flight plan

**Comment by okopanja** [comment]
Meanwhile: 2.8 update did break entry into PVI, needs investigation...

**Comment by seven2099** [comment]
Good work! - let me know if there's any help I can provide.

**Comment by okopanja** [comment]
> Good work! - let me know if there's any help I can provide.
@seven2099 I made updates and made unofficial 1.6.1 release to support both BS2, BS3. This will be the last update to this pull request and I will stop making the updates on this. 

To support BS properly with ABRIS, I made too many changes to the original project for this pull request to be accepted for merge. Instead, I believe I will have something **way cooler** to announce for **BS2** and **BS3** owners in near future... ;)

Perhaps in two weeks...

**Comment by okopanja** [comment]
The pull request will be closed in favor of dedicated Ka-50 waypoint entry tool here: https://github.com/okopanja/SharkPlanner

I will freeze the forked repository, but keep sources available for further references.

**Pull request closed without merging**